File: Proyecto1_1096917/DynamicStructures/EstrcuturasDinamicas/Node.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EstrcuturasDinamicas
{
    /// <summary>
    /// Estructura de datos básica. 
    /// </summary>
    /// <typeparam name="T">Nodo genérico, el parámetro será el elemento a insertar dependiendo de su tipo.</typeparam>
    public class Node<T>
    {
        private T element;//valor
        private Node<T> next;//nodo siguiente        

        /// <summary>
        /// Node. Constructor de la Clase.
        /// </summary>
        /// <param name="newElement">Elemento del nodo.</param>
        /// <param name="nextNode">Referencia a nodo siguiente.</param>
        public Node(T newElement, Node<T> nextNode)
        {
            element = newElement;
            next = nextNode;
        }

        /// <summary>
        /// getElement
        /// </summary>
        /// <returns>Retorna el elemento del nodo.</returns>
        public T getElement()
        {
            return element;
        }

        /// <summary>
        /// getNext
        /// </summary>
        /// <returns>Obtiene el nodo siguiente del nodo actual.</returns>
        public Node<T> getNext()
        {
            return next;
        }

        /// <summary>
        /// setNext
        /// </summary>
        /// <param name="next">Permite cambiar el nodo siguiente del nodo analizado.</param>
        public void setNext(Node<T> next)
        {
            this.next = next;
        }
    }
}

File: Final1096917/Final1096917/Node.cpp
#include "pch.h"
#include "Node.h"


Node::Node(int nTipo, Lavadora lava, Television tele, Node *nextNode) {	
	tipo = nTipo;
	if (tipo == 1) {//tipo 1 lavadora, tipo 2 television
		lavadora = lava;
	}
	else { 
		television = tele;
	}
	next = nextNode;
}


Electrodomestico Node::getElement() {
	if (tipo == 1) { return lavadora; }
	else { return television; }
}

Node* Node::getNext() {
	return next;
}

void Node::setNext(Node *next) {
	this->next = next;
}

Lavadora Node::getLavadora() { return lavadora; }

Television Node::getTelivision() { return television; }

int Node::getTipo() { return tipo; }

Node::~Node()
{
}